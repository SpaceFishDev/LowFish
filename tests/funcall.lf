i32 a()
{
	i32 x = 0;
	return x;
}
i32 b()
{
	a();
}
