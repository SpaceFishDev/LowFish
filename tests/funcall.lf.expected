<NODE( 'PROGRAM', PROGRAM )>
{
    <NODE( 'FUNCTION', FUNCTION )>
    {
        <NODE( 'i32', TYPE )>
        <NODE( 'a', TOKEN_NODE )>
        <NODE( 'BLOCK', BLOCK )>
        {
            <NODE( 'EXPRESSION', EXPRESSION )>
            {
                <NODE( 'VARDECL', VARDECL )>
                {
                    <NODE( 'i32', TYPE )>
                    <NODE( 'ASSIGNMENT', ASSIGNMENT )>
                    {
                        <NODE( 'x', TOKEN_NODE )>
                        <NODE( 'BASICEXPRESSION', BASICEXPRESSION )>
                        {
                            <NODE( '0', TOKEN_NODE )>
                        }
                    }
                }
            }
            <NODE( 'EXPRESSION', EXPRESSION )>
            {
                <NODE( 'x', RETURN )>
            }
        }
    }
    <NODE( 'FUNCTION', FUNCTION )>
    {
        <NODE( 'i32', TYPE )>
        <NODE( 'b', TOKEN_NODE )>
        <NODE( 'BLOCK', BLOCK )>
        {
            <NODE( 'EXPRESSION', EXPRESSION )>
            {
                <NODE( 'a', FUNCTION_CALL )>
            }
        }
    }
}

